### PR TITLE
[CLOUDSTACK-10276] listVolumes not working when storage UUID is not a UUID

### PR DESCRIPTION
**Pull request by rafaelweingartner**
## Description

When configuring a pre-setup primary storage we can enter the name-label of the storage that is going to be used by ACS and is already set up in the host. The problem is that we can use any String of characters there, and this String does not need to be a UUID. When listing volumes from a primary storage that has such conditions, the list will return all of the volumes in the cloud because the “API framework” will ignore that value as it is not a UUID type.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
Locally

## Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [X] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

File: api/src/org/apache/cloudstack/api/command/admin/volume/ListVolumesCmdByAdmin.java [hunk 1]
```diff
@@ -34,32 +34,5 @@
 @APICommand(name = "listVolumes", description = "Lists all volumes.", responseObject = VolumeResponse.class, responseView = ResponseView.Full, entityType = {Volume.class},
         requestHasSensitiveInfo = false, responseHasSensitiveInfo = false)
 public class ListVolumesCmdByAdmin extends ListVolumesCmd {
-    public static final Logger s_logger = Logger.getLogger(ListVolumesCmdByAdmin.class.getName());
-
-    @Parameter(name=ApiConstants.POD_ID, type=CommandType.UUID, entityType=PodResponse.class,
-            description="the pod id the disk volume belongs to")
-    private Long podId;
```
Comment (rhtyd) [inline api/src/org/apache/cloudstack/api/command/admin/volume/ListVolumesCmdByAdmin.java]: @rafaelweingartner Is podId an unused parameter?

Comment (rafaelweingartner) [reply api/src/org/apache/cloudstack/api/command/admin/volume/ListVolumesCmdByAdmin.java]: It is in the superclass as well. These parameters were declared twice. One in the "admin" class, and other in the "normal" one.

Comment (rhtyd) [reply api/src/org/apache/cloudstack/api/command/admin/volume/ListVolumesCmdByAdmin.java]: make sense to remove if parent class has them already declared.

**Comment by rafaelweingartner** [comment]
@rhtyd I thought about that. However, after evaluating the volume of changes, I decided to do a simple fix.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
arguing with jenkins

**Comment by DaanHoogland** [comment]
see if jenkins is convinced

**Comment by rhtyd** [comment]
I've my reservations on the fix, instead perhaps it may be better and could be simpler to handle use of external id as uuid when adding a pre-setup storage pool. This may not require us to remove the uuid-id conversations/annotations in the list API, also this will break auto-completion of storage ids in various APIs for CLIs such as cloudmonkey.

**Comment by rafaelweingartner** [comment]
@rhtyd how does it break cloudmonkey? 
Cloudmonkey continues to see the same thing. There is not change in the API, nothing will change for clients. I just stopped using the automagic conversion system via reflection that takes the UUID entered as ID (a UUID) and then converts to a database long ID.

**Comment by rhtyd** [comment]
@rafaelweingartner yes you're right, I tested it and it works due to a heuristic I had implemented (and forgotten about it, which uses the api arg removing id/ids and tries to find a list API). It's eod for me here, let me get back to you tomorrow if there can be any simpler fix.

**Comment by rhtyd** [comment]
I'm unable to find time to do the investigation on this. @DaanHoogland can you advise, and if it's all okay please go ahead with merging this.

**Comment by rafaelweingartner** [comment]
I really do not see any other smaller/simpler solution for this issue. Moreover, it does not affect/break anything in the client side. The client is already seeing an ID, which is a UUID.

The changes are only internal to ACS. I changed the type of `storageId` from `long` to `String` to avoid using that method that converts the given UUID to an internal database ID. That method is validating the input, and ignoring anything that is not an UUID. Then, I am using the UUID as the filter at `QueryManagerImpl.java`.

The changed lines are only 2. The other lines are a result of a code cleanup that I did, and the code formatting I applied.

**Comment by DaanHoogland** [comment]
@rafaelweingartner I am fine with merging this so far as we don't allow for manually entering a non-uuid storage id. In my understanding it doesn't. Is that correct?

**Comment by borisstoyanov** [comment]
@rhtyd, @DaanHoogland, @PaulAngus Cool, this seems to be the last pending item in the 4.11.1 milestone, I think we can merge and cut the RC.

**Comment by rafaelweingartner** [comment]
Actually, it does allow entering "non-UUID" parameters, and that is the problem I am addressing here. When adding a pre-setup storage, we use the field "name-label" from XenServer (e.g. "pre-setup storage 1"). If we set anything, but a UUID in that field, we can add a pre-setup storage with the name-label, and ACS will use the name-label as the UUID of the storage. 

This creates a problem, with the listVolumes method. Because we annotated the `storageId` as a UUID in the ACS code. There is a method, which intercepts the request and converts the UUIDs (String values) entered by users to database IDs, which are long values. The problem is that this method only handles UUIDs, anything else is ignored. Therefore, when listing volumes of a pre-setup storage, we get the volume listing without filtering by “storageId”.

By changing the `storageId` from `long` to `String`, and changing the annotation from `UUID` to `STRING`, we avoid using the method that converts String UUIDs into database long IDs, and everything starts working again.

Did you understand the problem and the solution I used?

**Comment by rhtyd** [comment]
Alright merging this now.

**Pull request merged**